x/blob/client/cli/payforblob.go

```diff
@@ -32,55 +32,79 @@ const (
-			"blob must be a hex encoded string of any length.\n" +
-			// TODO: allow for more than one blob to be sumbmitted via the CLI
-			"This command currently only supports a single blob per invocation.\n",
+		Short: "Pay for a data blobs to be published to Celestia.",
+		Long: "Pay for a data blobs to be published to Celestia.\n" +
+			`Where blob.json contains: 
+
+		{
+			"Blobs": [
+				{
+					"namespaceId": "0x00010203040506070809",
+					"blob": "0x48656c6c6f2c20576f726c6421"
+				},
+				{
+					"namespaceId": "0x00010203040506070809",
+					"blob": "0x48656c6c6f2c20576f726c6421"
+				}
+			]
+		}
+
+		namespaceID is the user-specifiable portion of a version 0 namespace. It must be a hex encoded string of 10 bytes.\n
+		blob must be a hex encoded string of any length.\n
+		
+		`,
 		Aliases: []string{"PayForBlob"},
 		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) < 2 {
-				return fmt.Errorf("PayForBlobs requires two arguments: namespaceID and blob")
+			if len(args) < 1 {
+				return fmt.Errorf("PayForBlobs requires one arguments: path to blob.json")
 			}
 			return nil
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			arg0 := strings.TrimPrefix(args[0], "0x")
-			namespaceID, err := hex.DecodeString(arg0)
-			if err != nil {
-				return fmt.Errorf("failed to decode hex namespace ID: %w", err)
-			}
+			clientCtx, err := client.GetClientTxContext(cmd)
 			namespaceVersion, err := cmd.Flags().GetUint8(FlagNamespaceVersion)
-			if err != nil {
-				return err
-			}
-			namespace, err := getNamespace(namespaceID, namespaceVersion)
-			if err != nil {
-				return err
-			}
+			path := args[0]
 
-			arg1 := strings.TrimPrefix(args[1], "0x")
-			rawblob, err := hex.DecodeString(arg1)
+			paresdBlobs, err := parseSubmitBlobs(clientCtx.Codec, path)
 			if err != nil {
-				return fmt.Errorf("failure to decode hex blob: %w", err)
+				return err
 			}
 
-			shareVersion, _ := cmd.Flags().GetUint8(FlagShareVersion)
-			blob, err := types.NewBlob(namespace, rawblob, shareVersion)
-			if err != nil {
-				return err
+			var blobs []*blob.Blob
+			for i := range paresdBlobs {
+				namespaceID, err := hex.DecodeString(strings.TrimPrefix(paresdBlobs[i].NamespaceId, "0x"))
+				if err != nil {
+					return fmt.Errorf("failed to decode hex namespace ID: %w", err)
+				}
+				namespace, err := getNamespace(namespaceID, namespaceVersion)
+				if err != nil {
+					return err
+				}
+				hexStr := strings.TrimPrefix(paresdBlobs[i].Blob, "0x")
+				rawblob, err := hex.DecodeString(hexStr)
+				if err != nil {
+					fmt.Printf("failure to decode hex blob value %s: %s", hexStr, err.Error())
+					continue
+				}
+
+				shareVersion, _ := cmd.Flags().GetUint8(FlagShareVersion)
+				blob, err := types.NewBlob(namespace, rawblob, shareVersion)
+				if err != nil {
+					fmt.Printf("failure to create blob with hex blob value %s: %s", hexStr, err.Error())
```

Comment:
[blocking]
as is would this submit a transaction without the blob that failed? if that's the case then I think we need to return with error. that could be immediatly or using something like multierror

Reply:
yes agreed will update this